src/channel_handler/types.rs

```diff
@@ -931,3 +931,55 @@ impl UnrollCoin {
         ))
     }
 }
+
+pub struct CoinDataForReward {
+    pub coin_string: CoinString,
+    // parent: CoinID,
+    // puzzle_hash: PuzzleHash,
+    // amount: Amount,
+}
+
+pub struct UnrollTarget {
+    pub state_number: usize,
+    pub unroll_puzzle_hash: PuzzleHash,
+    pub my_amount: Amount,
+    pub their_amount: Amount,
+}
+
+impl LiveGame {
+    /// Back up the live game state to the state we know so that we can generate the puzzle
```

Comment:
I would use the word "Copy" instead of "back up". "Back up is a little ambiguous - before I read further, I thought you meant "go to a previous state", but now I see you mean "grab a copy".

Reply:
But it's literally about, if necessary, returning to an earlier state.  I would say 'backup' if what i meant was storing the state.

Reply:
Maybe 'regress' is better.

Reply:
also note this part of the code is badly in flux.  i have no idea if this particular thing will survive.  i'm experimenting with ideas around being able to figure out what i'm looking at as much as actually writing code that does the thing, although hopefully I'll be able to solidify that idea relatively soon.

Reply:
maybe should have waited to make this pr.